### ISSUE TITLE
Latest emacs build is blocked due to "missing publisher"

### ISSUE THREAD

**Issue by pank** [kiennq/scoop-misc#13]
Sorry, this is related to https://github.com/kiennq/emacs-build/ but that repo has no issue tracker. 
This repo seemed somewhat related...

I use your [emacs builds](https://github.com/kiennq/emacs-build) on my PC. They are great so thanks!
I can't build on my windows PC and I don't have an admin account.

Something happened with the latest version and it gets blocked. Windows claims it does not have a published, but it's [probably something else ](https://stackoverflow.com/questions/15358185/changing-publisher-information-for-a-exe-file) (I don't see a published for the previous version either...).

![image](https://github.com/kiennq/scoop-misc/assets/666845/281606a7-a1b6-4217-8a53-96d4b92ffe67)

This [version](https://github.com/kiennq/emacs-build/releases/tag/v30.210.20231001.4a8f100) is blocked on my PC. 

The version from ‎19 ‎September ‎2023 works, I guess this [one](https://github.com/kiennq/emacs-build/releases/tag/v30.209.20230914.71a875f), works well.

Sorry again for using the wrong issue tracker.

**Comment by kiennq** [comment]
It's working for me though.
Can you try the latest one to see if it's working for you?
The exe is built by Github action so it should be safe, if you check the hash is the same as the one in the scoop then I guess you can probably allow it to run.

**Comment by pank** [comment]
I downloaded [the latest release](https://github.com/kiennq/emacs-build/releases/tag/v30.212.20231007.0b64a8f).
Still the same. 

I probably have a more locked-down windows environment than most.
I don't have admin rights and I cannot allow or disallow something to run. But the version that was probably compiled on Sep. 14 works. I think it is [this one](https://github.com/kiennq/emacs-build/releases/tag/v30.209.20230914.71a875f).

Could the [change in the build setup](https://github.com/kiennq/emacs-build/commit/b98216fd243a5a2067321e82cf32e1b92f255d9b#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R47) have changed something about exe signatures? I.e. this: 
```
      - name: Setup Msys2
        uses: msys2/setup-msys2@v2
        if: runner.os == 'Windows'
        with:
          msystem: UCRT64
          update: true
          release: false
```
Was `UCRT64` used before?

For now, it seems that copying over `runemacs.exe` from the old installation makes it run...

**Comment by pank** [comment]
Aside and almost totally unrelated.

Since [a while](https://github.com/kiennq/emacs-build/commit/45cdd016598c745096b88db465bfc813238c2f7f#diff-39f8e1a51c829e09d28bbbc01ace7e6b1eb9484c9b55f5ec941cc019fe853677R409), the [`lisp/leim/quail`](https://git.savannah.gnu.org/cgit/emacs.git/tree/lisp/leim/quail) folder has been missing from the package. Apparently it's part of the `slim` packaging.

This breaks `set-input-method`, `toggle-input-method` and `activate-transient-input-method`. The folder is less than a mb.

**Comment by kiennq** [comment]
> Aside and almost totally unrelated.
> 
> Since [a while](https://github.com/kiennq/emacs-build/commit/45cdd016598c745096b88db465bfc813238c2f7f#diff-39f8e1a51c829e09d28bbbc01ace7e6b1eb9484c9b55f5ec941cc019fe853677R409), the [`lisp/leim/quail`](https://git.savannah.gnu.org/cgit/emacs.git/tree/lisp/leim/quail) folder has been missing from the package. Apparently it's part of the `slim` packaging.
> 
> This breaks `set-input-method`, `toggle-input-method` and `activate-transient-input-method`. The folder is less than a mb.

Yeah, I intentionally omit that one as I think we should use Windows's input method instead of uncompleted support in Emacs. The `leim` folder is around 20MB ish and cutting it will trim down Emacs a bit.

**Comment by kiennq** [comment]
> Was `UCRT64` used before?

The UCRT64 is using the new ucrt version of stdlibc++ instead of the old msvcrt. This is the recommendation now so I would like to keep it.
Btw, I've submitted the runemacs to MS malware asking for a rescan and unblock from smartscreen, let's see if that will resolve your issue

**Comment by pank** [comment]
> Yeah, I intentionally omit that one as I think we should use Windows's input method instead of uncompleted support in Emacs. The leim folder is around 20MB ish and cutting it will trim down Emacs a bit.

I just like the TeX input method as it's the quickest way to write unicode math. I don't know much about what it possible with windows input methods. Never checked. 

Re size: I just looked at the numbers on the savannah and summed them up (829084) and assumed they were bytes and that they wouldn't change much during compilation. But I didn't check after compilation sizes.

**Comment by pank** [comment]
> The UCRT64 is using the new ucrt version of stdlibc++ instead of the old msvcrt. This is the recommendation now so I would like to keep it.

This seems reasonable. 

> Btw, I've submitted the runemacs to MS malware asking for a rescan and unblock from smartscreen, let's see if that will resolve your issue

There must be some magic metadata missing and I guess that could be it...

**Comment by pank** [comment]
Turns out that you can make it work by calling the powershell command `unblock-file` on the `exe` files.

**Comment by kiennq** [comment]
Thank you, I will close this issue then,
